src/java/org/apache/cassandra/config/DataStorage.java

```diff
@@ -0,0 +1,385 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.config;
+
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import com.google.common.primitives.Ints;
+
+/**
+ * Represents an amount of data storage. Wrapper class for Cassandra configuration parameters, providing to the
+ * users the opportunity to be able to provide config with a unit of their choice in cassandra.yaml as per the available
+ * options. (CASSANDRA-15234)
+ */
+public final class DataStorage
+{
+    /**
+     * The Regexp used to parse the storage provided as String.
+     */
+    private static final Pattern STORAGE_UNITS_PATTERN = Pattern.compile("^(\\d+)(kb|KB|mb|MB|gb|GB|b|B)$");
```

Comment:
could be simplified by just toLowerCase

Reply:
Probably if we use it with `locale.ROOT`, similar to what you did in the tests?
On the other hand I was looking that in cql3/Duration and in types/Duration classes we refrain from that so I kept it consistent. What do you think?

Reply:
> Probably if we use it with locale.ROOT, similar to what you did in the tests?

WFM

Reply:
On the other hand, it looks like these are going to change to `KiB`, `MiB`, etc. In that case, I don't see a problem with being strict on the casing.

Reply:
Yeah, in the world of scientific units, casing tends to matter. (ex. M vs. m?)

Reply:
Then I guess I should do the same for Duration if we want to be consistent